plugin-server/src/cdp/hog-transformations/hog-transformer.service.ts

```diff
@@ -145,40 +148,24 @@ export class HogTransformerService {
 
                 const shouldRunHogWatcher = Math.random() < this.hub.CDP_HOG_WATCHER_SAMPLE_RATE
 
-                // Get states for all functions to check if any are disabled - only if feature flag is enabled
-                let states: Record<string, { state: HogWatcherState }> = {}
-                if (shouldRunHogWatcher) {
-                    try {
-                        const timer = transformEventHogWatcherLatency.startTimer({ operation: 'getStates' })
-                        states = await this.hogWatcher.getStates(teamHogFunctions.map((hf) => hf.id))
-                        timer()
-                    } catch (error) {
-                        logger.warn('⚠️', 'HogWatcher getStates failed', { error })
-                    }
-                }
-
                 for (const hogFunction of teamHogFunctions) {
                     const transformationIdentifier = `${hogFunction.name} (${hogFunction.id})`
 
-                    // Check if the function would be disabled via hogWatcher - but don't actually disable yet
                     if (shouldRunHogWatcher) {
-                        const functionState = states[hogFunction.id]
-                        if (functionState?.state >= HogWatcherState.disabledForPeriod) {
-                            // Just log that we would have disabled this transformation but we're letting it run for testing
+                        // Check if function is in a degraded state
+                        let functionState = this.getHogFunctionState(hogFunction.id)
+                        if (!functionState) {
+                            functionState = await this.handleCacheMiss(hogFunction.id)
+                        }
+
+                        // If the function is in a degraded state, skip it
+                        // If no state is found, we will continue with the transformation
+                        if (functionState && functionState >= HogWatcherState.disabledForPeriod) {
                             logger.info(
```

Comment:
Can we swap this to a metric? Logs are already really noisy and hard to easily query on